### DISCUSSION TITLE
autoincrement for type number

### DISCUSSION THREAD

**Discussion by pgswow** [pocketbase/pocketbase#1038] [Q&A] [answered]
Hello!
Perhaps it's already been asked whether it's possible to add an auto-increment property to the number type.
At the moment I am doing this:

> try {
>       const number = await locals.pb.collection('tickets').getFullList(99999, {sort: '-number'})
>       await locals.pb.collection('tickets').create({title, description, author: locals.user.id, number: number[0].number += 1})
>     } catch (error) {
>       console.log(error)
>     }

Is there a way to make this easier?

**Comment by ganigeorgiev** [comment, marked as answer]
No there is no other way at the moment (unless you use PocketBase as framework and handle the autoincrement in a `OnRecordBeforeCreateRequest` hook).

There is an issue for something similar in - https://github.com/pocketbase/pocketbase/issues/786.

**Comment by bmingles** [comment]
@ganigeorgiev are there any examples of how the `OnRecordBeforeCreateRequest` hook could be used to mimic `AUTOINCREMENT` behavior for a numeric field?
